nectar/downloaders/threaded.py

```diff
@@ -273,119 +273,128 @@ def _fetch(self, request, session):
+
+                    bytes_read = len(chunk)
+                    report.bytes_downloaded += bytes_read
+
+                    now = datetime.datetime.now()
+
+                    if now - last_update_time >= progress_interval:
+                        last_update_time = now
+                        self.fire_download_progress(report)
+
+                    with self._bytes_lock:
+                        if now - self._time_bytes_this_second_was_cleared >= ONE_SECOND:
+                            self._bytes_this_second = 0
+                            self._time_bytes_this_second_was_cleared = now
+                        self._bytes_this_second += bytes_read
+
+                    if max_speed is not None and self._bytes_this_second >= max_speed:
+                        # it's not worth doing fancier mathematics than this, very
+                        # fine-grained sleep times [1] are not honored by the system
+                        # [1] for example, sleeping the remaining fraction of time
+                        # before this second is up
+                        time.sleep(0.5)
+
+                # guarantee 1 report at the end
+                self.fire_download_progress(report)
+
+            except SkipLocation:
+                _logger.debug("Skipping {url} because {netloc} could not be reached.".format(
+                    url=request.url, netloc=netloc)
+                )
+                report.download_skipped()
+
+            except requests.ConnectionError as e:
+                if nretry < DEFAULT_TRIES - 1:
+                    continue
+                _logger.error(_('Skipping requests to {netloc} due to repeated connection'
+                                ' failures: {e}').format(netloc=netloc, e=str(e)))
+                self.failed_netlocs.add(netloc)
+                report.download_connection_error()
+
+            except requests.Timeout:
+                if nretry < DEFAULT_TRIES - 1:
+                    continue
+                """
+                Handle a timeout differently than a connection error. Do not add
+                to failed_netlocs so that a new connection can be attempted.
+                """
+                _logger.warning("Request Timeout - Connection with {url} timed out.".format(
+                    url=request.url)
+                )
+                report.download_connection_error()
+
+            except DownloadCancelled as e:
+                _logger.info(str(e))
+                report.download_canceled()
+
+            except DownloadFailed as e:
+                if nretry < DEFAULT_TRIES - 1:
+                    continue
+                _logger.info('Download failed: %s' % str(e))
+                report.error_msg = e.args[2]
+                report.error_report['response_code'] = e.args[1]
+                report.error_report['response_msg'] = e.args[2]
+                report.download_failed()
+
+            except Exception as e:
```

Comment:
i don't feel comfortable to have 5 tries no matter what exception got caught

Reply:
so how many retries should I use? 3?

Reply:
my question was around mostly why have any retries for anything expect connection error?

Reply:
In my opinion, 2 times should be enough. Otherwise, it might take too long to fail if the server is really busy and can't really accept take more connections.

Reply:
2 retries would also be ok w/ me.